lib/jekyll/filters/url_filters.rb

```diff
@@ -23,14 +22,23 @@ def absolute_url(input)
       # Returns a URL relative to the domain root as a String.
       def relative_url(input)
         return if input.nil?
-        site = @context.registers[:site]
-        return ensure_leading_slash(input.to_s) if site.config["baseurl"].nil?
+        return ensure_leading_slash(input.to_s) if sanitized_baseurl.nil?
         Addressable::URI.parse(
-          ensure_leading_slash(site.config["baseurl"]) + ensure_leading_slash(input.to_s)
+          ensure_leading_slash(sanitized_baseurl) + ensure_leading_slash(input.to_s)
         ).normalize.to_s
       end
 
       private
+
+      def site
+        @context.registers[:site]
+      end
+
+      def sanitized_baseurl
+        baseurl = site.config["baseurl"]
+        return baseurl unless baseurl.nil? || (baseurl == "/")
```

Comment:
> Will this fail and not return if baseurl is nil? 

Ruby will `return nil` if `baseurl` equals `nil` or `/`

> Can we compress trailing slashes to a single one, or removed entirely, as required by the concatenation?

IMO substituting a regex match with something is not necessary as it is known to affect performance.  But again, there is no issue in employing that method if there's a strong use case. Setting `baseurl` with multiple slashes is wrong.
The least appropriate measure would be to output a warning message that the `baseurl` has been configured incorrectly.